src/com/facebook/buck/android/MergeAndroidResourcesStep.java

```diff
@@ -345,26 +351,62 @@ void writePerPackageRDotJava(
       }
 
       String packageName = entry.getValue();
+
+      List<RDotTxtEntry> resources = new ArrayList<>();
+      SortedSetMultimap<String, RDotTxtEntry> styleableIntArrToIntsMap = TreeMultimap.create();
+
       for (String line : linesInSymbolsFile) {
         Optional<RDotTxtEntry> parsedEntry = RDotTxtEntry.parse(line);
         Preconditions.checkState(parsedEntry.isPresent(), "Should be able to match '%s'.", line);
 
-        // We're only doing the remapping so Roboelectric is happy and it is already ignoring the
-        // id references found in the styleable section.  So let's do that as well so we don't have
-        // to get fancier than is needed.  That is, just re-enumerate all app-level resource ids
-        // and ignore everything else, allowing the styleable references to be messed up.
         RDotTxtEntry resource = parsedEntry.get();
+        resources.add(resource);
+        if (resource.idType == IdType.INT && resource.type == RType.STYLEABLE) {
+          String attrName = resource.name.substring(0, resource.name.indexOf('_'));
+          styleableIntArrToIntsMap.put(attrName, resource);
```

Comment:
styleableIntArrToIntsMap is a multimap so not needed, we expect multiple values.